test/unit.test.js

```diff
@@ -0,0 +1,109 @@
+import { 
+    getServiceClientName, 
+    getServiceClientDescription, 
+    getClientClassName, 
+    getFunctionName, 
+    getFunctionDetails, 
+    getSendOperations, 
+    getReceiveOperations 
+} from '../components/helpers/utils'; 
+
+describe('Service Client Helpers', () => {
+    const mockAsyncAPI = {
+        info: jest.fn(() => ({
+            title: jest.fn(() => 'TestService'),
+            description: jest.fn(() => 'Test Service Description'),
+            hasDescription: jest.fn(() => true)
+        })),
+    };
+
+    const mockAsyncAPIWithoutDesc = {
+        info: jest.fn(() => ({
+            title: jest.fn(() => 'TestService'),
+            description: jest.fn(() => undefined),
+            hasDescription: jest.fn(() => false)
+        })),
+    };
+
+    const mockOperations = [
+        { 
+            operationId: jest.fn(() => 'sendTestOperation'), 
+            id: jest.fn(() => 'receiveTestOperation'),
+            channels: jest.fn(() => [{ address: jest.fn(() => 'test/topic') }]), 
+            summary: jest.fn(() => 'Test Operation Summary'),
+            isSend: jest.fn(() => true),
+            isReceive: jest.fn(() => false),
+        },
+        {
+            operationId: jest.fn(() => 'receiveTestOperation'),
+            id: jest.fn(() => 'receiveTestOperation'),
+            channels: jest.fn(() => [{ address: jest.fn(() => 'test/topic') }]),
+            summary: jest.fn(() => 'Test Operation Summary'),
+            isSend: jest.fn(() => false),
+            isReceive: jest.fn(() => true),
+        }
+    ];
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('getServiceClientName', () => {
+        it('should return the correct service client name', () => {
+            const result = getServiceClientName(mockAsyncAPI);
+            expect(result).toEqual('TestService Client');
+        });
+    });
+
+    describe('getServiceClientDescription', () => {
+        it('should return the correct service client description', () => {
+            const result = getServiceClientDescription(mockAsyncAPI);
+            expect(result).toEqual('Test Service Description');
+        });
+
+        it('should return empty string if no description provided', () => {
+            const result = getServiceClientDescription(mockAsyncAPIWithoutDesc);
+            expect(result).toEqual('');
+        });
+    });
+
+    describe('getClientClassName', () => {
+        it('should return the correct client class name', () => {
+            const result = getClientClassName(mockAsyncAPI);
+            expect(result).toEqual('TestServiceClient');
+        });
+    });
+
+    describe('getFunctionName', () => {
+        it('should return the correct function name for a given operation', () => {
+            const result = getFunctionName(mockOperations[0]);
```

Comment:
`getFunctionName` needs refactor

use `hasOperationId()` like you did in other function with `hasDescription()`

Reply:
we need to handle `undefined` case, what use as a filename if operationId is undefined:

- current way `return str.split('/').map(word => word.charAt(0).toLowerCase() + word.slice(1)).join('');` is to get some string (channel name) and make it ready for filename, we can also use `filenamify`
- basically throw an error to the user that generation is not possible without operation id as we need to know how to name functions, and we do it basing on the operation id